pkg/collector/corechecks/gpu/nvmlmetrics/collector.go

```diff
@@ -0,0 +1,154 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+// Package nvmlmetrics holds the logic to collect metrics from the NVIDIA Management Library (NVML).
+// The main entry point is the Collector struct, which is responsible for collecting metrics from the NVML library.
+// There are multiple subsystems that can be used to collect metrics from NVML: GPM, FieldInfo functions, and general
+// API functions. The collection of metrics is done by the subsystems, implementing the subsystemCollector interface.
+package nvmlmetrics
```

Comment:
There's another file called `nvml.go`, I used this to specify only the retrieval of metrics from NVML was here. Depending on how things end up after the PRs moving things I don't know if that `nvml` file is still going to be in the core check or not.

Reply:
i meant rename the pkg to nvml instead of nvmlmetrics

Reply:
so lets rename it to nvidia pkg, and put the nvml under that pkg as well. similar to containers check with different container runtimes (cri, containerd), we will create sub-packages based on gpu vendor.

in the future we might add extra layer of packages inside each vendor pkg, but i think it is not necessary atm.

WDYT?

Reply:
Looks good. Do you mind if we move that in a separate PR, to avoid conflicts? The system-probe refactor PR I think moves the `nvml.go` file and the draft PR with the NVML collectors also depends on this package being here, I feel it will be less work to move it after that. I can create a ticket to track that if you want.

Reply:
https://datadoghq.atlassian.net/browse/EBPF-597